### PR TITLE
Refactor webapi directory structure

### PR DESCRIPTION
**Pull request by dehoward**
### Motivation and Context

as the project has grown, the structure of webapi has become a bit hard to parse through, especially for a new dev coming into the project. this PR aims to reduce that tension and in general, reduce the clutter within the api.

> NOTE: the PR will be easier to review by going through each commit separately. 

### Description

- moves `deploy` folder into `scripts` folder.
- moves `importdocument` folder into `tools` folder.
- moves code from `webapi/CopilotChat` up a level and removes duplicate folders.
- changes root namespace from `SemanticKernel.Service` to `CopilotChat.WebApi`.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [Contribution Guidelines](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by crickman — DISMISSED** [review]
Nice structureal update.  Love the shorter namespaces.

File: scripts/deploy/main.json [hunk 1]
```diff
@@ -380,7 +380,7 @@
             "value": "Warning"
           },
           {
-            "name": "Logging:LogLevel:SemanticKernel.Service",
+            "name": "Logging:LogLevel:CopilotChat",
```
Comment (momuno) [inline scripts/deploy/main.json:383]: CopilotChat or ChatCopilot?

File: scripts/deploy/package-webapi.sh [hunk 2]
```diff
@@ -72,7 +72,7 @@ if [[ ! -d "$PUBLISH_ZIP_DIRECTORY" ]]; then
 fi
 
 echo "Build configuration: $CONFIGURATION"
-dotnet publish "$SCRIPT_ROOT/../webapi/CopilotChatWebApi.csproj" --configuration $CONFIGURATION --framework $DOTNET --runtime $RUNTIME --self-contained --output "$PUBLISH_OUTPUT_DIRECTORY"
+dotnet publish "$SCRIPT_ROOT/../../webapi/CopilotChatWebApi.csproj" --configuration $CONFIGURATION --framework $DOTNET --runtime $RUNTIME --self-contained --output "$PUBLISH_OUTPUT_DIRECTORY"
 if [ $? -ne 0 ]; then
```
Comment (momuno) [inline scripts/deploy/package-webapi.sh:76]: general -- is there a plan to move the file naming to chat copilot?

Comment (teresaqhoang) [reply scripts/deploy/package-webapi.sh:76]: I don't think so but it makes sense to do that in this PR since it's already a big structure change to the webapi

File: webapi/CopilotChatWebApi.sln
Comment (teresaqhoang) [inline webapi/CopilotChatWebApi.sln:1]: This was an explicit addition to avoid VS annoyances since it'll automatically create one and offer to save it on spin up https://github.com/microsoft/semantic-kernel/pull/1885

File: webapi/CopilotChat/Models/DocumentImportForm.cs
Comment (teresaqhoang) [inline webapi/CopilotChat/Models/DocumentImportForm.cs:1]: Can you create subfolders for the models? They're growing exponentially and getting hard to parse through

**Review by teresaqhoang — CHANGES_REQUESTED** [review]

File: scripts/deploy/package-webapi.sh (hunk 2 above)
Comment (dehoward) [reply scripts/deploy/package-webapi.sh:76]: I'd like to keep these changes separate from the rename if you all are ok with it. 
I will follow-up this PR with another to rename things across the board (in both webapi and webapp): #102

File: scripts/deploy/main.json (hunk 1 above)
Comment (dehoward) [reply scripts/deploy/main.json:383]: CopilotChat, just for now until the rename is done (see below comment)

File: scripts/deploy/package-webapi.sh (hunk 2 above)
Comment (momuno) [reply scripts/deploy/package-webapi.sh:76]: makes sense to me!

Comment (teresaqhoang) [reply scripts/deploy/package-webapi.sh:76]: Same!

File: .gitignore [hunk 3]
```diff
@@ -287,6 +287,9 @@ FakesAssemblies/
 .ntvs_analysis.dat
 node_modules/
 
+# Visual Studio solution file
+*.sln
```
Comment (glahaye) [inline .gitignore:291]: Why do you want to ignore changes the the solution file?

File: webapi/CopilotChatWebApi.csproj [hunk 4]
```diff
@@ -5,7 +5,7 @@
     <LangVersion>10</LangVersion>
     <Nullable>enable</Nullable>
     <ImplicitUsings>disable</ImplicitUsings>
-    <RootNamespace>SemanticKernel.Service</RootNamespace>
+    <RootNamespace></RootNamespace>
```
Comment (glahaye) [inline webapi/CopilotChatWebApi.csproj:8]: If you are going to leave this empty, might as well remove the line completely

File: webapi/README.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Semantic Kernel Service - CopilotChat
+# Copilot Chat Web API
```
Comment (glahaye) [inline webapi/README.md:1]:
```suggestion
# Chat Copilot Web API
```

File: webapi/appsettings.json [hunk 6]
```diff
@@ -38,8 +38,8 @@
   //
   "AIService": {
     "Type": "AzureOpenAI",
-    "Endpoint": "", // ignored when AIService is "OpenAI"
-    // "Key": "",
+    "Endpoint": "https://lightspeed-team-shared-openai.openai.azure.com/", // ignored when AIService is "OpenAI"
+    "Key": "9e2fe86fbbfd4438bdc9bf81e8a1a9c3",
```
Comment (glahaye) [inline webapi/appsettings.json:42]: Dammit!  Now we have to regenerate out key :(

File: .gitignore (hunk 3 above)
Comment (glahaye) [reply .gitignore:291]: I see there were to SLN files.  Why not keep one of them?

**Review by glahaye — APPROVED** [review]
Wrong button

**Review by glahaye — CHANGES_REQUESTED** [review]
Remove those credentials!

File: webapi/appsettings.json [hunk 7]
```diff
@@ -39,7 +39,7 @@
   "AIService": {
     "Type": "AzureOpenAI",
     "Endpoint": "", // ignored when AIService is "OpenAI"
-    // "Key": "",
+    "Key": "",
```
Comment (glahaye) [inline webapi/appsettings.json:42]: Don't tempt fate! :)

```suggestion
    // "Key": "",
```

File: webapi/appsettings.json [hunk 8]
```diff
@@ -38,8 +38,8 @@
   //
   "AIService": {
     "Type": "AzureOpenAI",
-    "Endpoint": "https://lightspeed-team-shared-openai.openai.azure.com/", // ignored when AIService is "OpenAI"
-    "Key": "9e2fe86fbbfd4438bdc9bf81e8a1a9c3",
+    "Endpoint": "", // ignored when AIService is "OpenAI"
+    "Key": "",
```
Comment (momuno) [inline webapi/appsettings.json:42]: There can still be risk doing it this way because the creds will remain in the prior commit.  Can you revert the commit that originally added your creds instead of removing creds in additional commit? (I've been in this same boat before too)

Comment (glahaye) [reply webapi/appsettings.json:42]: Too late. We'll generate new keys instead.

File: webapi/README.md (hunk 5 above)
Comment (dehoward) [reply webapi/README.md:1]: will rename in separate PR, see https://github.com/microsoft/chat-copilot/pull/100#discussion_r1283601447

File: webapi/CopilotChatWebApi.csproj (hunk 4 above)
Comment (dehoward) [reply webapi/CopilotChatWebApi.csproj:8]: a warning is thrown when this is removed:
image

any ideas?

File: .gitignore (hunk 3 above)
Comment (dehoward) [reply .gitignore]: I removed it because it's not needed to run the project. but I've added it back

File: webapi/CopilotChatWebApi.csproj (hunk 4 above)
Comment (dehoward) [reply webapi/CopilotChatWebApi.csproj:8]: leaving blank. please lmk if you know of a better way to get around this warning

File: webapi/CopilotChatWebApi.sln
Comment (dehoward) [reply webapi/CopilotChatWebApi.sln:1]: added back

File: scripts/deploy/main.bicep [hunk 9]
```diff
@@ -271,7 +271,7 @@ resource appServiceWebConfig 'Microsoft.Web/sites/config@2022-09-01' = {
         value: 'Warning'
       }
       {
-        name: 'Logging:LogLevel:SemanticKernel.Service'
```
Comment (alliscode) [inline scripts/deploy/main.bicep:274]: These changes will need to coordinated with an update to the 'Deploy To Azure' links because they have a hard-coded path to main.json. Sync with me and we can update it at the same time.

File: webapi/CopilotChatWebApi.csproj (hunk 4 above)
Comment (alliscode) [reply webapi/CopilotChatWebApi.csproj:8]: I have a couple thoughts on this. Right now the directory structure has an extra CopilotChat directory in the webApi:

- /root (CopilotChat)
  - /webapi
       - /CopilotChat  <<-- Does this directory add any value?
           - /Models
           
If we get rid of that CopilotChat directory in the middle, then the RootNamespace would be "CopilotChat" and the namespace of a model could be CopilotChat.WebApi.Models.

**Review by alliscode — CHANGES_REQUESTED** [review]

**Review by anilgaddam0966 — COMMENTED** [review]
Good

File: webapi/CopilotChat/Models/DocumentImportForm.cs
Comment (dehoward) [reply webapi/CopilotChat/Models/DocumentImportForm.cs:1]: yup, good idea. added 3 subfolders for Request, Response, & Storage

File: CopilotChatWebApi.sln [hunk 10]
```diff
@@ -1,9 +1,9 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Version 17
 VisualStudioVersion = 17.5.33530.505
```
[the remark this answers is not in the document]
Comment (dehoward) [reply CopilotChatWebApi.sln]: yeah, that makes sense. updated to this structure.

File: webapi/CopilotChatWebApi.csproj (hunk 4 above)
Comment (dehoward) [reply webapi/CopilotChatWebApi.csproj]: I removed the folder but still had to manually set the `RootNamespace` to `CopilotChat.WebApi` to have the compiler recognize `CopilotChat.WebApi.Models` as correct

**Review by alliscode — DISMISSED** [review]

**Review by alliscode — APPROVED** [review]

File: scripts/deploy/main.bicep [hunk 11]
```diff
@@ -271,7 +271,7 @@ resource appServiceWebConfig 'Microsoft.Web/sites/config@2022-09-01' = {
...
+        name: 'Logging:LogLevel:CopilotChat.WebApi'
         value: 'Warning'
       }
```
Comment (glahaye) [inline scripts/deploy/main.bicep:276]: I notice you updated the bicep file but not the associated JSON ARM template.  Please synch both!

**Review by glahaye — APPROVED** [review]

**Review by teresaqhoang — APPROVED** [review]

File: scripts/deploy/main.bicep (hunk 11 above)
Comment (961527) [reply scripts/deploy/main.bicep:276]: - [ ] - image__[image]()
